### DISCUSSION TITLE
Add `repo-settings` scope to GITHUB_TOKEN in actions

### DISCUSSION THREAD

**Discussion by thetillhoff** [community/community#64547] [Actions]
### Select Topic Area

Product Feedback / Feature Request

### Body

Repository settings become a hassle to manage when you have more than just a few.

Currently, it's possible to make bulk adjustments via API calls with curl or even terraform.

But if the GITHUB_TOKEN for github actions could be given a scope similar to `repo-settings`, where the workflow/job/step can be allowed to write to the settings of the repo, it'd be possible to
- store the repo settings in a file -> infrastructure-as-code, versioning of settings, "blaming" (git blame) of changes
- store the repo settings "set-command" in github actions, for example with the `gh` cli tool (which doesn't support files at the moment, but we have to start somewhere ;) )
- the repo settings can be set automatically via github actions without workarounds like private access tokens

The default should be `NONE` so it has to be enabled explicitly for the desired step if this is desired and to prevent bad things from happening.
Difference to the `Administration` scope of the fine-grained PATs is that it doesn't allow repo creation nor deletion.

---

Just a thought, but probably harder to implement:
Even cooler would be a `.github/repo-settings.yaml` (or similar) which is automatically enforced and settings are grayed out if they are set there. That would reduce the headache of how to keep the settings properly in sync via pipelines for example when someone changed something manually by accident.

**Comment by travi** [comment]
On a somewhat related note, it sounds like you are trying to accomplish something similar to the goals of https://github.com/repository-settings/app, so I thought I'd share in case you weren't aware of it. Full disclosure, I'm the primary maintainer of that project, but have no direct relationship with GitHub, so I can't provide insight for your primary question.

**Reply by thetillhoff** [reply]
Thanks @travi, that's pretty what I'm referring to. Nice to see I'm not the only one thinking in that direction.

Sadly it's not from a verified publisher, which means I can't use it in my case (corporate setting).
I opened https://github.com/repository-settings/app/issues/783 to address this issue.
